packages/eslint-plugin/src/lib/rules/type-annotation-spacing.ts

```diff
@@ -26,7 +26,7 @@ const defaultOptions = [
   {}
 ];
 
-module.exports = {
+const rule: Rule.RuleModule = {
```

Comment:
eslint-plugin-eslint-plugin is going to stop working

Reply:
Can we submit an enhancement request to eslint-plugin-eslint-plugin to use static variable analysis tracking in order to avoid requiring that `module.exports` be directly assigned with the rule object?

That is, this should be equally valid for the plugin:

```js
const rule = { /* ... */ };

module.exports = rule;
```

Reply:
what do you think about 
```
export default {
  meta: {},
  create(context) {
  }
} as Rule.RuleModule
```

Reply:
looking at their code, it doesn't matter how we type it - it's specifically looking for `module.exports =`, `module.exports.[create | meta]` or `exports.[create | meta]`.
So switching to es6 export breaks the module, regardless of the typing!

https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/blob/a9a97b2d907888a846d4b57fb36beca83267f651/lib/utils.js#L47-L50

https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/blob/master/lib/utils.js#L71-L77

https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/blob/a9a97b2d907888a846d4b57fb36beca83267f651/lib/utils.js#L81-L85